Dockerfile

```
@@ -1,59 +1,48 @@
-FROM registry.access.redhat.com/ubi9/ubi-minimal:9.4@sha256:c0e70387664f30cd9cf2795b547e4a9a51002c44a4a86aa9335ab030134bf392 as base-python
+#
+# Base image with defaults for all stages
+FROM registry.access.redhat.com/ubi9/python-312@sha256:88ea2d10c741f169681102b46b16c66d20c94c3cc561edbb6444b0de3a1c81b3 AS base
+
 COPY LICENSE /licenses/LICENSE
-ENV PYTHONUNBUFFERED=1
-
-RUN microdnf upgrade -y && \
-    microdnf install -y \
-        shadow-utils \
-        python3.11 && \
-    microdnf clean all && \
-    update-alternatives --install /usr/bin/python3 python /usr/bin/python3.11 1 && \
-    ln -snf /usr/bin/python3.11 /usr/bin/python && \
-    python3 -m ensurepip
-
-WORKDIR /code
-RUN useradd -u 5000 app && chown -R app:app /code
-
-# ---- Interims image to get and build all the python dependencies ----
-FROM base-python as build-base-python
-ENV POETRY_HOME=/opt/poetry \
-    POETRY_VIRTUALENVS_CREATE=false \
-    PIP_DISABLE_PIP_VERSION_CHECK=on \
-    PATH=/opt/poetry/bin:$PATH
-
-# install poetry
-ARG POETRY_VERSION
-RUN POETRY_VERSION=$POETRY_VERSION python3 - < <(curl -sSL https://install.python-poetry.org) && poetry --version
-
-# required for poetry to install the dependencies
-COPY --chown=app:app pyproject.toml poetry.lock README.md /code/
-# the code
-COPY --chown=app:app gitlab_project_exporter /code/gitlab_project_exporter/
-COPY --chown=app:app app.sh /code/
-
-# install the python dependencies
-RUN poetry install --no-interaction --no-ansi --only main
-
-# ---- Test image ----
-FROM build-base-python as test
-# install the test dependencies
-RUN microdnf install -y make
-RUN poetry install --no-interaction --no-ansi
-# install the tests
-COPY --chown=app:app tests /code/tests/
-COPY --chown=app:app Makefile /code/
-USER app:app
+
+#
+# Builder image
+#
+FROM base AS builder
+COPY --from=ghcr.io/astral-sh/uv:0.5.11@sha256:0ac957607303916420297a4c9c213bb33fbd3c888f9cd7f4f7273596ebf42b85 /uv /bin/uv
```

Please use the latest `uv` version.

```suggestion
COPY --from=ghcr.io/astral-sh/uv:0.5.15@sha256:ea861f8e28b5c0e85ec14dc0f367d9d5cfa5b418024cc250219288d4fff591f1 /uv /bin/uv
```